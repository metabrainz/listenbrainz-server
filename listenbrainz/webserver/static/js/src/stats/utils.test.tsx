import { mount } from "enzyme";

import getEntityLink from "./utils";

describe("getEntityLink", () => {
  it("renders a link if MBID is provided", () => {
    const wrapper = mount(getEntityLink("artist", "test_artist", "test_mbid"));
    expect(wrapper).toMatchSnapshot();
  });

  it("renders a string if MBID is not provided", () => {
    const wrapper = mount(getEntityLink("artist", "test_artist"));
    expect(wrapper).toMatchSnapshot();
  });
});
