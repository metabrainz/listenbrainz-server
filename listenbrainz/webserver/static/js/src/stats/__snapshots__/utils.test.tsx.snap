// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getEntityLink renders a link if MBID is provided 1`] = `
<a
  className="underlined-link"
  href="https://musicbrainz.org/artist/test_mbid"
  rel="noopener noreferrer"
  target="_blank"
>
  test_artist
</a>
`;

exports[`getEntityLink renders a string if MBID is not provided 1`] = `"test_artist"`;
